packages/smooth_app/lib/helpers/background_task_helper.dart

```diff
@@ -0,0 +1,181 @@
+import 'dart:convert';
+import 'dart:io';
+import 'package:openfoodfacts/openfoodfacts.dart';
+import 'package:openfoodfacts/utils/CountryHelper.dart';
+import 'package:smooth_app/database/dao_product.dart';
+import 'package:smooth_app/database/local_database.dart';
+import 'package:smooth_app/query/product_query.dart';
+import 'package:task_manager/task_manager.dart';
+
+const String IMAGE_UPLOAD_TASK = 'Image Upload';
+const String PRODUCT_EDIT_TASK = 'Product Edit';
+Future<TaskResult> callbackDispatcher(LocalDatabase localDatabase) async {
+  await TaskManager().init(
+      runTasksInIsolates: false,
+      executor: (Task inputData) async {
+        final String processName = inputData.data!['processName'] as String;
+        switch (processName) {
+          case IMAGE_UPLOAD_TASK:
+            return uploadImage(inputData.data!, localDatabase);
+
+          case PRODUCT_EDIT_TASK:
+            return otherDetails(inputData.data!, localDatabase);
+
+          default:
+            return TaskResult.success;
+        }
+      },
+      listener: (Task task, TaskStatus status) {});
+  return TaskResult.success;
+}
+
+Future<TaskResult> otherDetails(
+    Map<String, dynamic> inputData, LocalDatabase localDatabase) async {
+  final BackgroundOtherDetailsInput inputTask =
+      BackgroundOtherDetailsInput.fromJson(inputData);
+  final Map<String, dynamic> mp =
+      json.decode(inputTask.inputMap) as Map<String, dynamic>;
+  final User user =
+      User.fromJson(jsonDecode(inputTask.user) as Map<String, dynamic>);
+  await OpenFoodAPIClient.saveProduct(
+    user,
+    Product.fromJson(mp),
+    language: LanguageHelper.fromJson(inputTask.languageCode),
+    country: CountryHelper.fromJson(inputTask.country),
+  );
+  final DaoProduct daoProduct = DaoProduct(localDatabase);
+  final ProductQueryConfiguration configuration = ProductQueryConfiguration(
+    inputTask.barcode,
+    fields: ProductQuery.fields,
+    language: LanguageHelper.fromJson(inputTask.languageCode),
+    country: CountryHelper.fromJson(inputTask.country),
+  );
+
+  final ProductResult queryResult =
+      await OpenFoodAPIClient.getProduct(configuration);
+  if (queryResult.status == 1) {
+    final Product? product = queryResult.product;
+    if (product != null) {
+      await daoProduct.put(product);
+      localDatabase.notifyListeners();
+    }
+  }
+  // Returns true to let platform know that the task is completed
+  return TaskResult.success;
```

Comment:
What about errors? Do you only rely on the library?

Reply:
Yeah, Should I explicitly handle it here, 
I had done earlier but removed it since the task was already being handled with taskManager